packages/strapi-connector-bookshelf/lib/queries.js

```diff
@@ -650,52 +655,41 @@ const buildSearchQuery = (qb, model, params) => {
   const query = params._q;
 
   const associations = model.associations.map(x => x.alias);
+  const stringTypes = ['string', 'text', 'uid', 'email', 'enumeration', 'richtext', 'password'];
+  const numberTypes = ['biginteger', 'integer', 'decimal', 'float'];
 
-  const searchText = Object.keys(model._attributes)
-    .filter(attribute => attribute !== model.primaryKey && !associations.includes(attribute))
-    .filter(attribute => ['string', 'text'].includes(model._attributes[attribute].type));
-
-  const searchInt = Object.keys(model._attributes)
-    .filter(attribute => attribute !== model.primaryKey && !associations.includes(attribute))
-    .filter(attribute =>
-      ['integer', 'decimal', 'float'].includes(model._attributes[attribute].type)
-    );
-
-  const searchBool = Object.keys(model._attributes)
-    .filter(attribute => attribute !== model.primaryKey && !associations.includes(attribute))
-    .filter(attribute => ['boolean'].includes(model._attributes[attribute].type));
+  const searchColumns = Object.keys(model._attributes)
+    .filter(attribute => !associations.includes(attribute))
+    .filter(attribute => stringTypes.includes(model._attributes[attribute].type));
 
   if (!_.isNaN(_.toNumber(query))) {
-    searchInt.forEach(attribute => {
-      qb.orWhere(attribute, _.toNumber(query));
-    });
+    const numberColumns = Object.keys(model._attributes)
+      .filter(attribute => !associations.includes(attribute))
+      .filter(attribute => numberTypes.includes(model._attributes[attribute].type));
+    searchColumns.push(...numberColumns);
   }
 
-  if (query === 'true' || query === 'false') {
-    searchBool.forEach(attribute => {
-      qb.orWhere(attribute, _.toNumber(query === 'true'));
-    });
+  if ([...numberTypes, ...stringTypes].includes(model.primaryKeyType)) {
+    searchColumns.push(model.primaryKey);
   }
 
   // Search in columns with text using index.
   switch (model.client) {
-    case 'sqlite3': {
-      searchText.forEach(attr => qb.orWhereRaw(`${attr} LIKE ?`, `%${query}%`));
+    case 'pg':
+      searchColumns.forEach(attr =>
+        qb.orWhereRaw(`"${attr}"::text ILIKE ?`, `%${escapeQuery(query, '*%\\')}%`)
```

Comment:
I don't understand the escaping logic you have setup ?what is it supposed to do ?

Reply:
The escape function is here to avoid special characters in the query to impact how the query is made. Since "%", "_" and "\" are interpreted by the database.